portals/admin/source/src/app/components/AdvancedSettings/TenantConfSave.jsx

```diff
@@ -0,0 +1,240 @@
+    const intl = useIntl();
+    const [initialState, setInitialState] = useState({
+        tenantConf: '',
+        tenantConfSchema: '',
+    });
+    const [state, dispatch] = useReducer(reducer, initialState);
+    const {
+        tenantConf, tenantConfSchema,
+    } = state;
+    const restApi = new API();
+
+    useEffect(() => {
+        let tenantConfVal;
+        let tenantConfSchemaVal;
+        restApi.tenantConfSchemaGet().then((result) => {
+            tenantConfSchemaVal = result.body;
+            dispatch({ field: 'tenantConfSchema', value: tenantConfSchemaVal });
+        });
+        restApi.tenantConfGet().then((result) => {
+            tenantConfVal = JSON.stringify(result.body, null, '\t');
+            dispatch({ field: 'tenantConf', value: tenantConfVal });
+        });
+        setInitialState({
+            tenantConf: tenantConfVal,
+            tenantConfSchema: tenantConfSchemaVal,
+        });
+    }, []);
+
+    const editorWillMount = (monaco) => {
+        const schemaVal = state.tenantConfSchema;
+        monaco.languages.json.jsonDefaults.setDiagnosticsOptions({
+            completion: true,
+            validate: true,
+            format: true,
+            schemas: [{
+                uri: 'http://myserver/foo-schema.json',
+                fileMatch: ['*'],
+                schema: schemaVal,
+            }],
+        });
+    };
+
+    const tenantConfOnChange = (newValue) => {
+        dispatch({ field: 'tenantConf', value: newValue });
+    };
+
+    const cancelUpdate = () => {
+        let tenantConfVal;
+        restApi.tenantConfGet().then((result) => {
+            tenantConfVal = JSON.stringify(result.body, null, '\t');
+            const editState = {
+                tenantConf: tenantConfVal,
+            };
+            dispatch({ field: 'editDetails', value: editState });
+        });
+        setInitialState({
+            tenantConf: tenantConfVal,
+        });
+    };
+
+    const formSaveCallback = () => {
+        const tenantConfiguration = state.tenantConf;
+        const promisedUpdateTenantConf = restApi.updateTenantConf(tenantConfiguration);
+        return promisedUpdateTenantConf
+            .then(() => {
+                Alert.success(
+                    <FormattedMessage
+                        id='Settings.Advanced.TenantConf.edit.success'
+                        defaultMessage='Advanced Configuration saved successfully'
+                    />,
+                );
+                history.push('/settings/advanced');
+            })
+            .catch((error) => {
+                const { response } = error;
+                if (response.body) {
+                    Alert.error(response.body.description);
+                }
+                return null;
+            })
+            .finally(() => {
+                updateList();
+            });
+    };
+
+    return (
+        <ContentBase
+            pageStyle='full'
+            title={intl.formatMessage({
```

Comment:
Should have use `<FormattedMessage/>` component

Reply:
resolved